src/vm/methodtablebuilder.cpp

```diff
@@ -1809,7 +1811,7 @@ MethodTableBuilder::BuildMethodTableThrowing(
 
     if (IsValueClass())
     {
-        if (bmtFP->NumInstanceFieldBytes != totalDeclaredFieldSize || HasOverLayedField())
+        if (bmtFP->NumInstanceFieldBytes != totalDeclaredFieldSize || HasOverLayedField() || hasFloatingPointNumberField)
             GetHalfBakedClass()->SetIsNotTightlyPacked();
```

Comment:
@jkotas I tried to define a new bit for `HasFloatingPointNumberField()` but seems like we already exhausted the available bits. You can see all VMFLAG_*:
https://github.com/dotnet/coreclr/blob/44f57065649af5f8bcbb7c71d827221a7bc1bf7a/src/vm/class.h#L2103

Or should I define another enum struct?